File: .ProgramingBasics/.ComplexConditionalStatements/06.PointOnRectangleBorder/src/PointOnRectangleBorder.java
import java.util.Scanner;

public class PointOnRectangleBorder {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double x1 = Double.parseDouble(scanner.nextLine());
        double y1 = Double.parseDouble(scanner.nextLine());

        double x2 = Double.parseDouble(scanner.nextLine());
        double y2 = Double.parseDouble(scanner.nextLine());

        double x = Double.parseDouble(scanner.nextLine());
        double y = Double.parseDouble(scanner.nextLine());

        boolean isOnLeftSide = (x == x1) && (y >= y1 && y <= y2);
        boolean isOnRightSide = (x == x2) && (y >= y1 && y <= y2);
        boolean isOnTopSide = (y == y1) && (x >= x1 && x <= x2);
        boolean isOnBottomSide = (y == y2) && (x >= x1 && x <= x2);

        if (isOnLeftSide || isOnRightSide || isOnTopSide || isOnBottomSide) {
            System.out.println("Border");
        }
        else {
            System.out.println("Inside / Outside");
        }
    }
}

File: ProgramingBasics/ComplexConditionalStatements/06.PointOnRectangleBorder/06.PointOnRectangleBorder/PointOnRectangleBorder.cs
using System;

namespace _06.PointOnRectangleBorder
{
    public class PointOnRectangleBorder
    {
        public static void Main(string[] args)
        {
            var x1 = double.Parse(Console.ReadLine());
            var y1 = double.Parse(Console.ReadLine());

            var x2 = double.Parse(Console.ReadLine());
            var y2 = double.Parse(Console.ReadLine());

            var x = double.Parse(Console.ReadLine());
            var y = double.Parse(Console.ReadLine());

            var isOnLeftSide = (x == x1) && (y >= y1 && y <= y2);
            var isOnRightSide = (x == x2) && (y >= y1 && y <= y2);
            var isOnTopSide = (y == y1) && (x >= x1 && x <= x2);
            var isOnBottomSide = (y == y2) && (x >= x1 && x <= x2);

            if (isOnLeftSide || isOnRightSide || isOnTopSide || isOnBottomSide)
            {
                Console.WriteLine("Border");
            }
            else
            {
                Console.WriteLine("Inside / Outside");
            }
        }
    }
}
